CMake/kwiver-depends-Boost.cmake

```diff
@@ -3,6 +3,9 @@ if (NOT DEFINED KWIVER_BOOST_VERSION)
   set(KWIVER_BOOST_VERSION 1.54)
 endif()
 
+if (NOT ${KWIVER_BUILD_SHARED})
```

Comment:
I think the decision about this block has less to do with whether kwiver is built static and more to do with whether Boost was.  It think we will need something in the fletchConfig.cmake (if it's not already there) that says which way Boost was built.

Reply:
Moved the change to fletch in https://github.com/Kitware/fletch/pull/556